lib/fluent/plugin/out_amplitude.rb

```diff
@@ -189,22 +207,47 @@ def send_to_amplitude(records)
         )
         log.info("sent #{records.length} to amplitude")
       else
-        error_string = log_error(
+        error_string = error_postsend_fail(
           code: res.response_code,
           body: res.body,
           records: records,
-          duration: res.total_time * 1000
+          duration: res.total_time * 1000,
+          user_id_counts: user_id_counts.to_s
         )
         raise AmplitudeError, "Error: #{error_string}"
       end
     end
 
-    def log_error(error)
+    def error_postsend_fail(code, body, records, duration, user_id_counts)
```

Comment:
I'm not sure this interface matches what you have above—if I'm reading it correctly, a single hash is being passed in, but you're expecting five different arguments here.  Did you test this locally already?

Reply:
Has it been changed?  I see five arguments being passed into `error_postsend_fail`